### ISSUE TITLE
There's no way to disable and persist telemetry per project

### ISSUE THREAD

**Issue by flaviovs** [withastro/astro#11820]
### Astro Info

```block
Astro                    v4.14.4
Node                     v18.19.0
System                   Linux (x64)
Package Manager          npm
Output                   static
Adapter                  none
```

### If this issue only occurs in one browser, which browser is a problem?

n/a

### Describe the Bug

According to https://astro.build/telemetry/:

> You can always opt-out by running astro telemetry disable in the root of any Astro project directory. This will disable telemetry across your entire machine, not just the project directory that you run it in

Also:

> You can also opt-out by setting the environment variable: ASTRO_TELEMETRY_DISABLED=1

The above requires every developer to change their local environment to disable telemetry **before** running the CLI, which is problematic for compliance and/or privacy reasons for some orgs in regulated sectors.

### What's the expected result?

A setting in the SCM-managed config file to disable telemetry **for that specific project**, so new developers and/or scripts checking out the code don't break an org's regulations because they forgot to set up their local machines.

### Link to Minimal Reproducible Example

n/a

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Can you not do this via npm scripts?

```json
"scripts": {
  "dev": "ASTRO_TELEMETRY_DISABLED=1 astro dev",
  "build": "ASTRO_TELEMETRY_DISABLED=1 astro dev",
}
```

**Comment by flaviovs** [comment]
Yes, we can.

However NPM is not the CLI.

The documentation alone tells people to run `astro` (or `npx astro`) in many places.

**Comment by Fryuni** [comment]
You can set it on your config file:
```ts
// astro.config.mjs

process.env.ASTRO_TELEMETRY_DISABLED = '1';

export default defineConfig({ ... });
```

Running with debug shows:
```
astro:telemetry [record] telemetry has been disabled
```

**Comment by flaviovs** [comment]
That's a great solution, thanks for the tip.

BTW, this is what the CLI shows here, even when telemetry has been disabled using env variables/config change: 
```
$ DEBUG="astro:telemetry" npx astro build
▶ Astro collects anonymous usage data.
  This information helps us improve Astro.
  Run "astro telemetry disable" to opt-out.
  https://astro.build/telemetry

  astro:telemetry [notify] telemetry has been enabled +0ms
  astro:telemetry [record] telemetry has been disabled +1s
```

I'm okay with that, but just FYI this may still trigger compliance and audit issues in some orgs.

**Comment by Fryuni** [comment]
Yeah, technically, doing this leaves the telemetry enabled for tracking before the config file is loaded, that is what it notifies that it is enabled first.

But no telemetry is collected before the config is loaded (the relevant information comes from the config), so disabling it there disables all telemetry in Astro. Not a perfect solution, but it is a solution.

**Comment by bluwy** [comment]
Another option is to use the `"postinstall"` script:

```json
"scripts": {
  "postinstall": "astro telemetry disable"
}
```

**Comment by ascorbic** [comment]
Closing as there are several ways to handle this

**Comment by Feuermurmel** [comment]
> Closing as there are several ways to handle this

There is no reliable way to disable telemetry for a project. I want to prevent people working on a project I'm starting to unknowingly have telemetry collected from their machines.

I'm evaluating different static site generators for a new project. The current state of this is a definitive no-go.

**Comment by waynesbrain** [comment]
@Feuermurmel Does setting the `process.env` vars in `astro.config.mjs` no longer work? I was just about to try that out because this sucks:

```json
{ "start": "ASTRO_DISABLE_UPDATE_CHECK=false ASTRO_TELEMETRY_DISABLED=1 PROXY=1 astro dev",
  "start-lan": "ASTRO_DISABLE_UPDATE_CHECK=false ASTRO_TELEMETRY_DISABLED=1 LAN=1 PROXY=1 astro dev --host",
  "start-lan-proxy": "ASTRO_DISABLE_UPDATE_CHECK=false ASTRO_TELEMETRY_DISABLED=1 LAN=1 PROXY=1 astro dev --host",
  "start-proxy": "ASTRO_DISABLE_UPDATE_CHECK=false ASTRO_TELEMETRY_DISABLED=1 PROXY=1 astro dev"
}
```

**Comment by mstrauss** [comment]
Astro is now "owned" by Cloudflare it seems. Time to take it back.

**Comment by waynesbrain** [comment]
This is working for me, at the top of my `astro.config.mjs` files. You don't have to set `ASTRO_DISABLE_UPDATE_CHECK` to `false` like the docs say, you just have to set it to `!== "true"`, looking at the code.

```js
/**
 * @type {{
 * LAN?: string; PROXY?: string; PUBLIC_HOST_URL?: string; NODE_ENV?: string;
 * ASTRO_TELEMETRY_DISABLED?: string; ASTRO_DISABLE_UPDATE_CHECK?: string;
 * }}
 */
const env = process.env;
env.ASTRO_TELEMETRY_DISABLED = "1";
env.ASTRO_DISABLE_UPDATE_CHECK = "1";

// https://astro.build/config
export default defineConfig({ ... });
```
